mitxgraders/formulagrader/matrixgrader.py

```diff
@@ -58,4 +90,78 @@ def check_response(self, answer, student_input, **kwargs):
                 raise
             else:
                 return {'ok': False, 'msg': err.message, 'grade_decimal': 0}
+        except InputTypeError as err:
```

Comment:
Re old comment:
> Can we move this try/except block into MatrixGrader, so that comparers can just raise a ShapeError and it's treated appropriately? Open to suggestions here.

I moved the try/except block out of the comparer, but in order to do what we want (if I understood correctly) we need a new error type. Reason is that ShapeErrors happen for two reasons:

- Student input contains shape errors like `"[[1, 2, 3]] - 2`
- Author encounters a shape error when subtracting `comparer_params[0] - student_input`. In this case it would be confusing to student to see "Cannot add/subtract matrix and scalar", since student doesn't "see" any subtraction happening. So these errors are re-raised as InputTypeErrors.